sabnzbd/urlgrabber.py

```diff
@@ -59,6 +60,27 @@
 )
 
 
+def filename_from_content_disposition(content_disposition):
+    """
+    Extract and validate filename from a Content-Disposition header.
+
+    Origin: https://github.com/httpie/httpie/blob/4c8633c6e51f388523ab4fa649040934402a4fc9/httpie/downloads.py#L98
+    :param content_disposition: Content-Disposition value
+    :type content_disposition: str
+    :return: the filename if present and valid, otherwise `None`
+    :example:
+        filename_from_content_disposition('attachment; filename=jakubroztocil-httpie-0.4.1-20-g40bd8f6.tar.gz')
+        should return: 'jakubroztocil-httpie-0.4.1-20-g40bd8f6.tar.gz'
+    """
+    msg = Message(f'Content-Disposition: attachment; {content_disposition}')
+    filename = msg.get_filename()
+    if filename:
+        # Basic sanitation.
```

Comment:
Just to note (also for @jcfp) is that later this is passed through `sanitize_filename` so we are safe enough.